### ISSUE TITLE
MissingSuperCall reports overrides containing teardown logic

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#9528]
### Expected behavior

`MissingSuperCall` recognizes an explicit `super.onDestroyView()` call in a Fragment override.

### Observed behavior

Detekt `2.0.0-alpha.5` reports `MissingSuperCall` for overrides shaped like this:

```kotlin
override fun onDestroyView() {
    collaborator.releaseViewReferences()
    super.onDestroyView()
}
```

The super call is present and intentionally happens after releasing view-bound collaborators.

This reproduced in multiple AndroidX Fragment and Activity subclasses during type-resolved Android analysis.

### Context

- Detekt: `2.0.0-alpha.5`
- AndroidX Fragment/Activity
- `allRules = true`
- Type-resolved Android/Kotlin analysis

**Comment by BraisGabin** [comment]
Does the issue appear if you change the order? I recently fixed a related issue but it was not related with the order but if the annotation was in the grandparwnt instead of the parent.

**Comment by oaksprout** [comment]
Would you want this one done? — Fix a detekt false positive where the `MissingSuperCall` rule flags an override that legitimately doesn't call `super` because it's teardown/cleanup logic, not a bug. If yes, we'll do it and open a PR; if not, say so (or just ignore this) and we won't touch it.

How it works: an AI agent does the work against your write-up; it's independently checked in a clean container (your test suite, network off) before anything reaches you; a human reviews it, submits it, and answers the review. You keep full control — nothing lands unless you merge it.

Recent examples of the same flow on other projects: https://github.com/stretchr/testify/pull/1932, https://github.com/pypa/pip-audit/pull/1089, https://github.com/sigstore/sigstore-python/pull/1845.

We're testing whether work offered this way is useful to maintainers — a blunt "no" is as helpful as a yes.
